### ISSUE TITLE
Angular2: Rendering data from Observables

### ISSUE THREAD

**Issue by webia1** [angular/angular#10293]
**I'm submitting a possible  [x ] bug report

Following interpolation

    {{ (__responseData | async)?.calculation | json }}

outputs following structure

    [
        {
            "gross": 26.625834,
            "net": 20.425833,
            "tax": 6.2000003
        }
    ]

How can I get gross?

    {{ (__responseData | async)?.calculation[0].gross }}

does not work, and the all the following trys don't work too:

    {{ (__responseData.calculation[0] | async)?.gross }}
    {{ (__responseData.calculation.[0].gross  | async) }}

What's the mistake?

Reproduction here: http://plnkr.co/edit/d2XJoLsXFXjOH7fSmhbz?p=preview

TypeScript Version 1.8.10

"@angular/common": "2.0.0-rc.2",
    "@angular/compiler": "2.0.0-rc.2",
    "@angular/core": "2.0.0-rc.2",
    "@angular/forms": "^0.1.0",
    "@angular/http": "2.0.0-rc.2",
    "@angular/platform-browser": "2.0.0-rc.2",
    "@angular/platform-browser-dynamic": "2.0.0-rc.2",
    "@angular/router": "3.0.0-alpha.6",
    "bootstrap": "^3.3.6",
    "core-js": "^2.4.0",
    "es6-module-loader": "^0.17.8",
    "ng2-bootstrap": "^1.0.17",
    "ng2-table": "^1.0.2",
    "rxjs": "5.0.0-beta.6",
    "systemjs": "0.19.26",
    "zone.js": "^0.6.12"

**Comment by rinukkusu** [comment]
Witnessing same behavior in RC.4.

**Comment by jcob-ross** [comment]
Hi, I'll just leave this dirty wokaround here.

```
@Pipe({name: 'getArrayElement'})
export class GetArrayElementPipe implements PipeTransform {
  transform(value: any[], index: number = 0): any {
    if (value && value.constructor && value.constructor === Array) {
      return value[index].gross;
    }
    return null;
  }
}

...

<div>
      {{(__responseData | async)?.calculation | getArrayElement  }}
</div>
```

**Comment by JurajMlich** [comment]
Try this: `{{ (__responseData.calculation | async)?[0]?.gross }}`

**Comment by rinukkusu** [comment]
@JurajMlich You can actually go and try it yourself in the [Plunker OP provided](http://plnkr.co/edit/d2XJoLsXFXjOH7fSmhbz?p=preview)

**Comment by gdi2290** [comment]
use pipes

http://plnkr.co/edit/HG1aU6?p=preview

```typescript
//our root app component
import {Component, Pipe} from '@angular/core';
import {HTTP_PROVIDERS, Http} from '@angular/http';
import 'rxjs/Rx';
import {Observable} from 'rxjs/Rx';

@Pipe({
  name: 'get'
})
class Get {
  transform(val, args) {
    if (val === null) return val;
    return val[args];
  }
}

@Pipe({
  name: 'first'
})
class First {
  transform(val, args) {
    if (val === null) return val;
    return val[0];
  }
}

@Component({
  selector: 'my-app',
  providers: [HTTP_PROVIDERS],
  pipes: [
    Get,
    First
  ],
  template: `
    <div>
      {{ __responseData | async | get:'calculation' | first | get:'gross' }}
    </div>
  `,
  directives: []
})
export class App {
  __responseData: Observable<any> = new Observable<any>();
  
  constructor(private _http: Http) {
    this.__responseData = this._http.get("./data.json")
                            .map(res => res.json());
  }
}
```

**Comment by sod** [comment]
You could use `*ngFor="let calculation of calculations | async"`

```typescript
@Component({
  selector: 'my-app',
  providers: [HTTP_PROVIDERS],
  template: `
    <div *ngFor="let calculation of calculations | async">
      Gross: {{ calculation.gross }}<br>
      Net: {{ calculation.net }}
    </div>
  `,
  directives: [NgFor]
})
export class App {
  calculations: Observable<{gross:number; net:number}[]>;

  constructor(private _http: Http) {
    this.calculations = this._http.get("./data.json")
                            .map(res => res.json().calculation)
  }
}
```

http://plnkr.co/edit/3ze70dYycQxNyXA286mX?p=preview

**Comment by gdi2290** [comment]
what @sod suggested is better

**Comment by webia1** [comment]
thank you very much Alexander @sod , I've [posted your answer here](http://stackoverflow.com/questions/38583807/angular2-rendering-data-from-observables), if you don't object. 

Can someone approve my initial question: Is it a bug or desired behavior?

**Comment by gdi2290** [comment]
using rx is preferred over having the template handling it

**Comment by pkozlowski-opensource** [comment]
Normally `{{ (__responseData | async)?.calculation[0].gross }}` should be enough _but_ to make it fully work you will need to fix that landed in maser via  4ec2a30. It will be part of rc.5. Till rc.5 is released you can use one of the work-arounds offered here.
